python/pyarrow/parquet.py

```diff
@@ -1252,6 +1355,95 @@ def _make_manifest(path_or_paths, fs, pathsep='/', metadata_nthreads=1,
     return pieces, partitions, common_metadata_path, metadata_path
 
 
+class _ParquetDatasetV2:
+    """
+    ParquetDataset shim using the Dataset API under the hood.
+    """
+    def __init__(self, path_or_paths, filesystem=None, filters=None,
+                 partitioning="hive", read_dictionary=None, buffer_size=None,
+                 memory_map=False, **kwargs):
+        import pyarrow.dataset as ds
+        import pyarrow.fs
+
+        # Raise error for not supported keywords
+        for keyword, default in [
+                ("schema", None), ("metadata", None),
+                ("split_row_groups", False), ("validate_schema", True),
+                ("metadata_nthreads", 1)]:
+            if keyword in kwargs and kwargs[keyword] is not default:
+                raise ValueError(
+                    "Keyword '{0}' is not yet supported with the new "
+                    "Dataset API".format(keyword))
+
+        # map old filesystems to new one
+        # TODO(dataset) deal with other file systems
+        if isinstance(filesystem, LocalFileSystem):
+            filesystem = pyarrow.fs.LocalFileSystem(use_mmap=memory_map)
```

Comment:
For a follow up ticket, maybe memory_map should default to None so that we can reconcile it with an instance of pa.fs.LocalFileSystem
```suggestion
            filesystem = pyarrow.fs.LocalFileSystem(use_mmap=memory_map)
        elif isinstance(filesystem, pyarrow.fs.LocalFileSystem):
            assert memory_map is None
```
?

Reply:
> maybe memory_map should default to None so that we can reconcile it with an instance of pa.fs.LocalFileSystem

Yeah, if we keep it as an option to the filesystem, that sounds as a good idea. But there also has been some discussion recently about not tying it to the filesytem.